File: sieve/sieve.py
import sys

def sieve(n):
    primes = [True] * (n+1)
    counter = 0
    for i in range(2,n):
        if primes[i]:
            counter = counter + 1
            for j in range(i*i, n, i):
                primes[j] = False

    return counter

if __name__ == "__main__":
    sieve(int(sys.argv[1]))

File: sieve/sieve.jl
function sieve(n::Int64)
    primes = repeat([true], n)
    primes[1], primes[2] = (false, false)
    for i = 2:Int(sqrt(n) + 1)-1
        if primes[i+1]
            for j = i*i:i:n-1
                primes[j+1] = false
            end
        end
    end
    return [i for i in 0:length(primes)-1 if primes[i+1]]
end

if abspath(PROGRAM_FILE) == @__FILE__
    sieve(parse(Int, sys.argv[2]))
end
